test/e2e/azureidentity/azureidentity.go

```diff
@@ -0,0 +1,190 @@
+func DeleteOnCluster(name, templateOutputPath string) error {
+	cmd := exec.Command("kubectl", "delete", "-f", path.Join(templateOutputPath, name+".yaml"), "--ignore-not-found")
+	util.PrintCommand(cmd)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "Failed to delete AzureIdentity from the Kubernetes cluster")
+	}
+
+	return nil
+}
+
+// GetClientID will return the client id of a user-assigned identity on Azure
+func GetClientID(resourceGroup, name string) (string, error) {
+	cmd := exec.Command("az", "identity", "show", "-g", resourceGroup, "-n", name, "--query", "clientId", "-otsv")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to get the clientID from the identity in Azure")
+	}
+
+	return strings.TrimSpace(string(out)), nil
+}
+
+// GetPrincipalID will return the principal id (objecet id) of a user-assigned identity on Azure
+func GetPrincipalID(resourceGroup, name string) (string, error) {
+	cmd := exec.Command("az", "identity", "show", "-g", resourceGroup, "-n", name, "--query", "principalId", "-otsv")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to get the principalID from the identity in Azure")
+	}
+
+	return strings.TrimSpace(string(out)), nil
+}
+
+// GetAll will return a list of AzureIdentity deployed on a Kubernetes cluster
+func GetAll() (*aadpodid.AzureIdentityList, error) {
+	cmd := exec.Command("kubectl", "get", "AzureIdentity", "-ojson")
+	util.PrintCommand(cmd)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get AzureIdentity from the Kubernetes cluster")
+	}
+
+	list := aadpodid.AzureIdentityList{}
+	err = json.Unmarshal(out, &list)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to unmarshall json")
+	}
+
+	return &list, nil
+}
+
+// WaitOnReaderRoleAssignment will block until the assignement of 'Reader' role to an identity is executed successfully
+func WaitOnReaderRoleAssignment(resourceGroup, name string) (bool, error) {
+	principalID, err := GetPrincipalID(resourceGroup, name)
+	if err != nil {
+		return false, err
+	}
+
+	// Need to tight poll the following command because principalID is not
+	// immediately available for role assignment after identity creation
+	readyChannel, errorChannel := make(chan bool, 1), make(chan error)
+	duration := 100 * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), duration)
+	defer cancel()
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				errorChannel <- errors.Errorf("Timeout exceeded (%s) while assigning 'Reader' role to identity on Azure", duration.String())
+			default:
+				cmd := exec.Command("az", "role", "assignment", "create", "--role", "Reader", "--assignee-object-id", principalID, "-g", resourceGroup)
+				_, err := cmd.CombinedOutput()
+				log.Printf("Tight poll command result: %s\n", err)
```

Comment:
Can you please make this message more informative such as " Retrying role assignment.." or some such message.